queue_processors/queue_processor/stomp_client.py

```diff
@@ -0,0 +1,98 @@
+# ############################################################################
+# Autoreduction Repository :
+# https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################
+"""
+This module deals with the updating of the database backend.
+It consumes messages from the queues and then updates the reduction run
+status in the database.
+"""
+import logging.config
+import sys
+import traceback
+from model.message.job import Message
+from queue_processors.queue_processor.handle_message import HandleMessage
+from queue_processors.queue_processor.handling_exceptions import \
+    InvalidStateException
+from queue_processors.queue_processor.settings import LOGGING
+from utils.clients.queue_client import QueueClient
+
+
+class StompClient:
```

Comment:
Not sure about this name, only because to me this suggests that it is one of the utils clients like database_client or queue_client. To me this class represents the routing of messages and the middle layer between queue_client and message processing. Open to counter arguments here, this is just the way I interpreted it

Reply:
So `queue_client`? 
I originally picked this name as the class layout (e.g. `on_message`) is specific to Stomp, yet there isn't really much to indicate that this abides to an API

Reply:
I've gone for `QueueListener` since `QueueClient` is already used by something else

Feel free to let me know if you have any other ideas for what to call it

Reply:
`QueueListener` sounds good